File: frontend/src/types/theater.ts
// Theater and Show related types
export interface Theater {
  id: string;
  name: string;
  city: string;
  address: string;
  pincode: string;
  latitude: number;
  longitude: number;
  facilities: string[];
  screens: Screen[];
}

export interface Screen {
  screenId: string;
  name: string;
  formats: string[];
  seatLayout: SeatLayout;
}

export interface SeatLayout {
  rows: number;
  seatsPerRow: number[];
  tierMapping: TierMapping;
}

export interface TierMapping {
  classicRows: number[];
  primeRows: number[];
  premiumRows: number[];
  vipRows: number[];
}

export interface Show {
  id: string;
  movieId: string;
  theaterId: string;
  screenId: string;
  showDate: string;
  showTime: string;
  format: string;
  language: string;
  basePrices: Record<string, number>;
  isActive: boolean;
  movieTitle: string;
  moviePosterUrl: string;
  theaterName: string;
  theaterCity: string;
}

File: backend/api/src/main/java/com/moviebooking/entity/supabase/Theater.java
package com.moviebooking.entity.supabase;

import lombok.*;

import jakarta.persistence.*;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;
import java.util.List;

@Entity
@Table(name = "theaters")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Theater {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    private String name;

    private String city;

    private String address;

    private String pincode;

    private Double latitude;

    private Double longitude;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(columnDefinition = "jsonb")
    private List<String> facilities;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(columnDefinition = "jsonb")
    private List<Screen> screens;

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Screen {
        private String screenId;
        private String name;
        private List<Movie.MovieFormat> formats;
        private SeatLayout seatLayout;
    }

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class SeatLayout {
        private Integer rows;
        private List<Integer> seatsPerRow;
        private TierMapping tierMapping;
    }

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class TierMapping {
        private List<Integer> classicRows;
        private List<Integer> primeRows;
        private List<Integer> premiumRows;
        private List<Integer> vipRows;
    }
}
